spyder/plugins/editor/extensions/snippets.py

```diff
@@ -700,17 +700,9 @@ def cursor_changed(self, line, col):
             self.inserting_snippet = False
             return
 
-        ignore_calls = {'undo', 'redo'}
         node, nearest_snippet, _ = self._find_node_by_position(line, col)
         if node is None:
-            stack = inspect.stack()
-            ignore = False
-            # Check if parent call was due to text update on codeeditor
-            # caused by a undo/redo call
-            for call in stack:
-                if call[3] in ignore_calls:
-                    ignore = True
-                    break
+            ignore = self.editor.is_undoing or self.editor.is_redoing
```

Comment:
How are race conditions managed with these variables? i.e., There could be the case where the editor flags are set to False and this slot might not be called at all

Reply:
If this is called from the main thread there is no race condition. Can this be called from another thread?

Reply:
Well, personally, I don't know much about Signal dispatching in Qt, so if you say that everything occurs sequentially, I trust you

Reply:
Usually, when a signal is emitted from the main thread, it is processed immediately. This is why the call inspection worked as well because this needs to be called directly to have “undo” in the stack.